controllers/common.go

```diff
@@ -661,3 +661,36 @@ func (r *VolumeSnapshotRestoreReconciler) setVSRQueue(vsr *volsnapmoverv1alpha1.
 
 	return true, nil
 }
+
+func GetPodSecurityContext(namespace string, sourcePVCName string, c client.Client) (*corev1.PodSecurityContext, error) {
+
+	podSC := corev1.PodSecurityContext{}
+	podList := corev1.PodList{}
+	if err := c.List(context.Background(), &podList, &client.ListOptions{Namespace: namespace}); err != nil {
+		return nil, err
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Spec.Volumes != nil {
+			po := getPVCPod(&pod, sourcePVCName)
+
+			if po != nil && po.Spec.SecurityContext != nil {
+				podSC = *po.Spec.SecurityContext
+				return &podSC, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+func getPVCPod(pod *corev1.Pod, pvcName string) *corev1.Pod {
```

Comment:
This function name and how its used is misleading.

It's being used to check if this pod does have the PVC with this name or not.
```suggestion
func podHasPVCName(pod *corev1.Pod, pvcName string) bool {
```